src/sessions/PageManager.ts

```diff
@@ -235,4 +260,21 @@ export class PageManager {
     private useCookies() {
         return navigator.cookieEnabled && this.config.allowCookies;
     }
+
+    /*
+    Parses the domain from the referrer, if it is available
+    */
+    private getDomainFromReferrer() {
+        if (document.referrer !== undefined || document.referrer !== '') {
+            const domainName = document.referrer.match(
+                '([A-Za-z]+://[^/]+)[/]?'
+            );
+
+            if (domainName) {
+                return domainName[1];
+            }
```

Comment:
1. Yes. However, will be using URL interface instead of regex for this. (See next revision) 
2. Excluded in next revision 
3. Next revision use of URL interface should handle localhost as well